### PR TITLE
ENH: add CudaCommon remote module

### PR DESCRIPTION
**Pull request by SimonRit**
Follows the discourse discussion https://discourse.itk.org/t/external-module-bypass-itkdata/4202. We (@LucasGandel and I) have extracted from [RTK](https://github.com/SimonRit/RTK) the CudaCommon module (with git history), added a test and setup a Linux CI with a self-hosted agent. When this PR is merged and released, the corresponding code will be removed from RTK. We shall quickly add a Windows self-hosted agent and the python package for a specific Cuda version. Compliance level has been set to 3 since it's been used and tested for a long time in RTK but I don't mind unticking some things and downgrading it to level 2. cc @Thibault-Pelletier
## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/Remote/CudaCommon.remote.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#--
+#-- ## Compliance level 5 star (AKA ITK main modules, or remote modules that could become core modules)
+#--   - [ ] Widespread community dependance
+#--   - [ ] Above 90% code coverage
+#--   - [ ] CI dashboards and testing monitored rigorously
+#--   - [ ] All requirements of Levels 4,3,2,1
+#--
+#-- ## Compliance Level 4 star (Very high-quality code, perhaps small community dependance)
+#--   - [x] Meets all ITK code style standards
+#--   - [x] No external requirements beyond those needed by ITK proper
```
Comment (dzenanz) [inline Modules/Remote/CudaCommon.remote.cmake:14]: I think this should be off, because CUDA is an external requirement not needed by main ITK library.

**Review by dzenanz — APPROVED** [review]
Awesome.

File: Modules/Remote/CudaCommon.remote.cmake (hunk 1 above)
Comment (SimonRit) [reply Modules/Remote/CudaCommon.remote.cmake]: Of course, sorry, I overlooked this.

**Review by thewtex — APPROVED** [review]
Excellent! :1st_place_medal: 

CC @aylward

**Pull request merged**